pyaedt/modules/solutions.py

```diff
@@ -1941,83 +1690,134 @@ def plot_2d_cut(
+                self.logger.error("Wrong format quantity")
+                return False
             curves.append([x, y, "{}={}".format(y_key, data[y_key][theta_idx])])
-        show_legend = True
-        if len(curves) > 15:
-            show_legend = False
-        return plot_2d_chart(
-            curves, xlabel=xlabel, ylabel=qty_str, title=title, snapshot_path=export_image_path, show_legend=show_legend
-        )
 
+        if show:
+            show_legend = True
+            if len(curves) > 15:
+                show_legend = False
+            if is_polar:
+                return plot_polar_chart(
+                    curves,
+                    xlabel=xlabel,
+                    ylabel=farfield_quantity,
+                    title=title,
+                    snapshot_path=export_image_path,
+                    show_legend=show_legend,
+                )
+            else:
+                return plot_2d_chart(
+                    curves,
+                    xlabel=xlabel,
+                    ylabel=farfield_quantity,
+                    title=title,
+                    snapshot_path=export_image_path,
+                    show_legend=show_legend,
+                )
+        else:
+            return curves
+
+    # fmt: off
     @pyaedt_function_handler()
     def polar_plot_3d(
-        self,
-        qty_str="RealizedGain",
-        phi_scan=0,
-        theta_scan=0,
-        title="3D Plot",
-        convert_to_db=True,
-        export_image_path=None,
+            self,
+            farfield_quantity="RealizedGain",
+            phi_scan=0,
+            theta_scan=0,
+            title="3D Plot",
+            format_quantity="dB10",
+            export_image_path=None,
+            show=True,
+            **kwargs
     ):
+        # fmt: on
         """Create a 3d plot of specified quantity.
 
         Parameters
         ----------
-        qty_str : str, optional
-            Quantity to plot. Default `"RealizedGain"`.
+        farfield_quantity : str, optional
+            Far field quantity to plot. The default is`"RealizedGain"`.
+            Available functions are: `"RealizedGain"`, `"RealizedGain_Theta"`, "`RealizedGain_Phi"`, `"rETotal"`,
+            `"rETheta"`, `"rEPhi"`.
         phi_scan : float, int, optional
-            Phi Scan Angle in degree. Default `0`.
+            Phi scan angle in degree. Default `0`.
         theta_scan : float, int, optional
-            Theta Scan Angle in degree. Default `0`.
+            Theta scan angle in degree. Default `0`.
         title : str, optional
             Plot title. Default `"3D Plot"`.
-        convert_to_db : bool, optional
-            Either if the quantity has to be converted in db or not. Default is `True`.
+        format_quantity : str, optional
+            Conversion data function.
+            Available functions are: `"dB10"`, `"dB20"`, "`abs"`, `"real"`, `"imag"`, `"norm"`, `"ang"`, `"and_deg"`.
```

Comment:
```suggestion
            Available functions are: ``"dB10"``, ``"dB20"``, ``"abs"``, ``"real"``,
            ``"imag"``, ``"norm"``, ``"ang"``, ``"and_deg"``.
```
I thought the "and" in the last function (``and_deg``) was a mistake earlier. I see that it is here too, so I may have been wrong. I won't change it here, but please check that all occurrences are correct.